### PR TITLE
Resizable CallView

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes [#15876](https://github.com/vector-im/element-web/issues/15876)

Peek 2021-03-03 17-12

### CODE REVIEW

**Review by dbkr — COMMENTED** [review]
lgtm, this will be super useful. Could someone who knows more about the resizers take a look at this as well?

File: src/components/views/voip/CallViewForRoom.tsx [hunk 1]
```diff
@@ -79,11 +79,50 @@ export default class CallViewForRoom extends React.Component<IProps, IState> {
...
+        this.props.resizeNotifier.startResizing();
+    }
+
+    onResize = () => {
+        this.props.resizeNotifier.notifyTimelineHeightChanged();
+    }
+
+    onResizeStop = () => {
+        this.props.resizeNotifier.stopResizing();
+    }
```
Comment (turt2live) [inline src/components/views/voip/CallViewForRoom.tsx:92]: I'm a little surprised the linter isn't punishing you for lack of semicolons on these function definitions. Also they should all be `private`.

**Review by turt2live — APPROVED** [review]
Resizer bit looks fine, sorry it took so long to review (I assumed Dave's avatar meant he was looking at it - oops). The little bit of linting/member access bits are all I'd ask to be fixed.

**Review by turt2live — APPROVED** [review]
marking approved for code changes

File: src/components/views/voip/CallViewForRoom.tsx (hunk 1 above)
Comment (SimonBrandner) [reply src/components/views/voip/CallViewForRoom.tsx]: Sorry for the missing `private`. Regarding the semicolons, I am pretty sure they aren't used in many places all over the codebase, so I kind of assumed they were not necessary... 

Btw, thanks for reviewing a bunch of my PRs lately!

Comment (turt2live) [reply src/components/views/voip/CallViewForRoom.tsx]: no worries on the member access - it's a thing that we should probably have a linter for but don't (event handlers rarely need to be public). 

For semicolons: I suspect no one set up the lint rule for it.

**Comment by niquewoodhouse** [comment]
Just to confirm please, this is to resize the callview _only_ when you're within the room, not the callview when its floating bottom right and you're not in that room?

**Comment by SimonBrandner** [comment]
> Just to confirm please, this is to resize the callview only when you're within the room, not the callview when its floating bottom right and you're not in that room?

Yes, this is only within the room, not the PIP call view

**Review by niquewoodhouse — APPROVED** [review]
This looks good to me.  It's using the same styles as resize elsewhere in the product, so makes total sense as a starting point to iterate after shipping, if necessary.   
Thanks for this improvement!

**Pull request merged**